wireguard-go/README.md

```diff
@@ -0,0 +1,39 @@
+This directory contains to patches that can be applied to wireguard-go to leverage wolfCrypt as a crypto engine.
+
+## Patches
+- `Wireguard-Go-wolfCrypt-Port.patch` : This patch simply ports the cryptography to use the go-wolfssl wrapper and wolfSSL. The underlying algorithms used in wireguard are the same.
+- `Wireguard-Go-FIPS-wolfCrypt-port.patch` : This patch ports wireguard to use FIPS 140-3 wolfCrypt and FIPS approved algorithms. Blake2s is replaced with SHA-256, Chaha20_Poly1305 with AES-GCM, and X25519 with ECC P-256.
+
+## Building
+First, build and install both wolfSSL and go-wolfssl following the instructions here: https://github.com/wolfSSL/go-wolfssl/blob/master/README.md. Make sure that wolfSSL is configured as shown below.
+```
+./configure --enable-sp="yes, 256"
+```
+
+If you're interested in the FIPS build, configure as shown below. Make sure that wolfCrypt tests pass, the normal flow for doing this can be seen below.
+```
+./configure --enable-fips=v5 --enable-sp="yes, 256"
+make
+./wolfcrypt/test/testwolfcrypt
+./fips-hash.sh
+make
+./wolfcrypt/test/testwolfcrypt
+sudo make install
+```
+
+Then clone wireguard-go and apply whichever patch you're intersted in. Note that these patches were developed and tested on top of the following wireguard-go commit https://github.com/WireGuard/wireguard-go/commit/12269c2761734b1.
+```
+git clone https://github.com/WireGuard/wireguard-go.git
+cd wireguard-go
+patch -p1 < ../Wireguard-Go-wolfCrypt-Port.patch
+```
+
+Finally, install go-wolfssl into the project and run the unit tests.
+```
+cd wireguard-go
+go get -u github.com/wolfssl/go-wolfssl 
+go mod edit -replace github.com/wolfssl/go-wolfssl=<path to your go-wolfssl directory>
+cd device
+go test
```

Comment:
add note about setting `LD_LIBRARY_PATH`  to `/usr/local/lib` on Linux, which is the default system installation dir of wolfSSL and isn't a library search location by default

`LD_LIBRARY_PATH=/usr/local/lib/ go test`

Reply:
Could you share the error here? I'd expect this to be covered by go-wolfssl herehttps://github.com/wolfSSL/go-wolfssl/blob/1307b3e186609d1f757cf710c3e29beea8893ce7/aes.go#L25

Reply:
```
$ go test
/tmp/go-build1711171579/b001/device.test: error while loading shared libraries: libwolfssl.so.42: cannot open shared object file: No such file or directory
exit status 127
FAIL	golang.zx2c4.com/wireguard/device	0.001s

```

Reply:
Huh, I guess not. Updated the documentation to reflect this.